### ISSUE TITLE
How To Work Offline

### ISSUE THREAD

**Issue by banter7702** [humbertogontijo/homeassistant-roborock#574]
Hi, sorry if this is kind of unrelated to the project.
I appreciate this integration so much, very powerful.

Sometime this week, I came to a realization that my Roborock Q Revo is pretty useless when my internet went down. Even with this integration. I thought it was going to work offline. Yes sure, pressing the button works, but it does full cleaning with mop+vac, which is inefficient, unlike when this integration is used or when routine is used.

I believe valetudo doesn't have support for Q Revo. So this is out if the question for the meantime.

Is there any way we could use our Roborock's full functionality without internet?

**Comment by Metehomea** [comment]
As is indicated in the Roborock integration documentation, offline functionality is NOT possible...

**Comment by Lash-L** [comment]
Yes, we are able to communicate locally for 99% of the queries. The vacuum itself has some logic hardcoded into it when it loses internet that causes it to be not usable when the internet goes down
